litmus/compile.ml

```diff
@@ -45,9 +45,12 @@ module Generic (A : Arch_litmus.Base)
       let pointer = CType.Pointer base
       let code_pointer = Pointer (Base "ins_t")
       let tag = Base "tag_t"
+      let array_ty sz = CType.Array ("int", sz)
+        (* is it right to assume int as the base type?*)
```

Comment:
This assumption limits us to only having checks by using 32 bit element sizes (e.g. we can safely `V0.4S`, but not `V0.16B`). This forces the user to change their final constraint to interpret the Neon register as having 32 bit lanes.

We could infer the element sizes from the number of elements in the vector. However, this change would be Neon specific, and therefore not ideal.